common/beerocks/bwl/shared/nl80211_client_impl.cpp

```diff
@@ -549,6 +549,103 @@ bool nl80211_client_impl::get_sta_info(const std::string &interface_name,
         });
 }
 
+bool nl80211_client_impl::get_survey_info(const std::string &interface_name,
+                                          std::vector<survey_info> &survey_info_list)
+{
+    static struct nla_policy survey_policy[NL80211_SURVEY_INFO_MAX + 1];
+    survey_policy[NL80211_SURVEY_INFO_FREQUENCY] = {NLA_U32, 0, 0};
+    survey_policy[NL80211_SURVEY_INFO_NOISE]     = {NLA_U8, 0, 0};
+
+    if (!m_socket) {
+        LOG(ERROR) << "Socket is NULL!";
+        return false;
+    }
+
+    // Get the interface index for given interface name
+    int iface_index = if_nametoindex(interface_name.c_str());
+    if (0 == iface_index) {
+        LOG(ERROR) << "Failed to read the index of interface " << interface_name << ": "
+                   << strerror(errno);
+
+        return false;
+    }
+
+    return m_socket.get()->send_receive_msg(
+        NL80211_CMD_GET_SURVEY, NLM_F_DUMP,
+        [&](struct nl_msg *msg) -> bool {
+            nla_put_u32(msg, NL80211_ATTR_IFINDEX, iface_index);
+
+            return true;
+        },
+        [&](struct nl_msg *msg) {
+            struct nlattr *tb[NL80211_ATTR_MAX + 1];
+            struct genlmsghdr *gnlh = static_cast<genlmsghdr *>(nlmsg_data(nlmsg_hdr(msg)));
+            struct nlattr *sinfo[NL80211_SURVEY_INFO_MAX + 1];
+            survey_info survey_info;
+
+            // Parse the netlink message
+            if (nla_parse(tb, NL80211_ATTR_MAX, genlmsg_attrdata(gnlh, 0), genlmsg_attrlen(gnlh, 0),
+                          NULL)) {
+                LOG(ERROR) << "Failed to parse netlink message!";
+                return;
+            }
+
+            if (!tb[NL80211_ATTR_SURVEY_INFO]) {
+                LOG(ERROR) << "Survey data missing!";
+                return;
+            }
+
+            // Parse nested survey info
+            if (nla_parse_nested(sinfo, NL80211_SURVEY_INFO_MAX, tb[NL80211_ATTR_SURVEY_INFO],
+                                 survey_policy)) {
+                LOG(ERROR) << "Failed to parse nested survey attributes!";
+                return;
+            }
+
+            survey_info.in_use = sinfo[NL80211_SURVEY_INFO_IN_USE];
+
+            if (sinfo[NL80211_SURVEY_INFO_FREQUENCY]) {
+                survey_info.frequency_mhz = nla_get_u32(sinfo[NL80211_SURVEY_INFO_FREQUENCY]);
+            } else {
+                LOG(DEBUG) << "NL80211_SURVEY_INFO_FREQUENCY attribute is missing";
+            }
+
+            if (sinfo[NL80211_SURVEY_INFO_NOISE]) {
+                survey_info.noise_dbm = (int8_t)nla_get_u8(sinfo[NL80211_SURVEY_INFO_NOISE]);
+            } else {
+                LOG(DEBUG) << "NL80211_SURVEY_INFO_NOISE attribute is missing";
```

Comment:
If this is missing, it will always be missing. So the log would get very noisy. So I think we should:
- use LOG_N_TIMES or maybe LOG_EVERY_N;
- (maybe) use TRACE instead of DEBUG.

Reply:
Done (I used TRACE)